### ISSUE TITLE
Crash when using highlighter with stroke stabilizer (and finish stroke) turned on

### ISSUE THREAD

**Issue by rolandlo** [xournalpp/xournalpp#4301]
(Please complete the following information, and then delete this line)

**Affects versions :**
 - OS: Linux
 - (Linux only) Desktop environment: Wayland
 - Which version of libgtk do you use 3.24.34
 - Version of Xournal++: latest master 
![grafik](https://user-images.githubusercontent.com/40485433/194840391-4081095b-20e8-4165-bb7a-51bfdc074456.png)
 - Installation method: PPA

**Describe the bug**
When using the highlighter with the stroke stabilizer turned on Xournal++ crashes (with signal 6) seemingly randomly after a few strokes. In the terminal I see that the assertion `p.z > 0.0` in `xoj::view::StrokeViewHelper::drawWithPressure` is not satisfied.

```
xournalpp: ./src/core/view/StrokeViewHelper.cpp:45: xoj::view::StrokeViewHelper::drawWithPressure(cairo_t*, const std::vector<Point>&, const LineStyle&, double)::<lambda(const Point&, const Point&)>: Zusicherung »p.z > 0.0« nicht erfüllt.

** (xournalpp:40806): WARNING **: 12:46:31.737: [Crash Handler] Crashed with signal 6
```

Crash log:
<details>

```term
Error: signal 6
[bt]: (0) build/xournalpp(+0x30bdfc) [0x5590bba54dfc]
[bt]: (1) /lib/x86_64-linux-gnu/libc.so.6(+0x3bcf0) [0x7f8145e3bcf0]
[bt]: (2) /lib/x86_64-linux-gnu/libc.so.6(pthread_kill+0x11b) [0x7f8145e9210b]
[bt]: (3) /lib/x86_64-linux-gnu/libc.so.6(gsignal+0x16) [0x7f8145e3bc46]
[bt]: (4) /lib/x86_64-linux-gnu/libc.so.6(abort+0xd7) [0x7f8145e227fc]
[bt]: (5) /lib/x86_64-linux-gnu/libc.so.6(+0x2271b) [0x7f8145e2271b]
[bt]: (6) /lib/x86_64-linux-gnu/libc.so.6(+0x33596) [0x7f8145e33596]
[bt]: (7) build/xournalpp(+0x3f3e55) [0x5590bbb3ce55]
[bt]: (8) build/xournalpp(_ZN3xoj4view16StrokeViewHelper16drawWithPressureEP6_cairoRKSt6vectorI5PointSaIS5_EERK9LineStyled+0x22a) [0x5590bbb3d0fd]
[bt]: (9) build/xournalpp(_ZNK3xoj4view10StrokeView4drawERKNS0_7ContextE+0x836) [0x5590bbb3ca82]
[bt]: (10) build/xournalpp(_ZN13StrokeHandler13drawSegmentToERK5Point+0x335) [0x5590bbc62a5d]
[bt]: (11) build/xournalpp(_ZN16StrokeStabilizer6Active17quadraticSplineToERKNS_5EventE+0x4ef) [0x5590bbc65add]
[bt]: (12) build/xournalpp(_ZN16StrokeStabilizer6Active14finalizeStrokeEv+0x73) [0x5590bbc655d7]
[bt]: (13) build/xournalpp(_ZN13StrokeHandler20onButtonReleaseEventERK17PositionInputDatad+0x7f) [0x5590bbc62cad]
[bt]: (14) build/xournalpp(_ZN11XojPageView20onButtonReleaseEventERK17PositionInputData+0xbc) [0x5590bbbcc41a]
[bt]: (15) build/xournalpp(_ZN15PenInputHandler9actionEndERK10InputEvent+0x1dd) [0x5590bbadd24b]
[bt]: (16) build/xournalpp(_ZN18StylusInputHandler10handleImplERK10InputEvent+0x44e) [0x5590bbae31ba]
[bt]: (17) build/xournalpp(_ZN20AbstractInputHandler6handleERK10InputEvent+0x71) [0x5590bbbfd44b]
[bt]: (18) build/xournalpp(_ZN12InputContext6handleEP9_GdkEvent+0x392) [0x5590bbad85a0]
```
</details>

**To Reproduce**
1. Open a new document
2. Use the stylus to draw with the highlighter either using the highlighter from the toolbar or via pen button
3. Wait for the crash

**Expected behavior**
No crash

**Screenshots of Problem**
Here are my stroke stabilizer settings.
![grafik](https://user-images.githubusercontent.com/40485433/194841029-e8f06fbe-7c02-4d7d-8754-0b9323a41cd5.png)
Trying various settings it seems that Xournal++ only crashes if the "Finish stroke" checkbox is activated.

**Additional context**
Possible suspect: #4137, commit https://github.com/xournalpp/xournalpp/commit/553588a4cf858d76b357322e40f50f750ca69794

**Comment by rolandlo** [comment]
@bhennion Could you please check if the assertion is at fault or the stroke finisher.

**Comment by bhennion** [comment]
Well, `xoj::view::StrokeViewHelper::drawWithPressure` should never ever be called for a highligher stroke (because the highligher does not take pressure into account). So the assert is legitimate.

I can reproduce, so let me investigate. I think my adding the assert in 553588a4 just made visible a previously invisible bug.

**Comment by bhennion** [comment]
I guess this is the issue...
https://github.com/xournalpp/xournalpp/blob/2809cb4fd1ce20723be688421f54223602ed22bf/src/core/control/tools/StrokeStabilizer.cpp#L130-L135

So the fix should be easy. PR to come

**Comment by rolandlo** [comment]
After #4302 being merged some `.xopp`-files either do not open any more (when opened from the file browser) or crash when reaching a certain page. One example:

```xml
<xournal creator="xournalpp 1.1.2+dev" fileversion="4">
<title>Xournal++ document - see </title>
<page width="595.276" height="841.89">
<background type="solid" color="white" style="graph" />
<layer>
<stroke tool="pen" ts="0" fn="" color="#002e99ff" width="1.41 0.49994156 0.54505829 0.56364716 0.57535126 0.58877655 0.60151337 0.61115204 0.61771408 0.62046799 0.62184494 0.62287766 0.62322189 0.62322189 0.62322189 0.62459885 0.62632004 0.62941818 0.63251633 0.63527023 0.63733566 0.63871262 0.63940109 0.63940109 0.63871262 0.63802414 0.6376799 0.6376799 0.63802414 0.63802414 0.63802414 0.63561447 0.6376799 0.64387619 0.64731857 0.65041672 0.65282639 0.6538591 0.65489182 0.65523605 0.65592453 0.65592453 0.65558029 0.65489182 0.65420334 0.65420334 0.65454758 0.6655632 0.67416916 0.6862175 0.69757736 0.70551636 0.71102417 0.71274536 0.7130896 0.7130896 0.7130896 0.71205688 0.70930298 0.6979216 0.58017059 0" capStyle="round">348.96909 148.79162 348.69433 148.63138 348.48832 148.40244 348.39673 148.10484 348.4654 147.76144 348.69433 147.46384 349.08355 147.16624 349.63298 146.84574 350.274 146.66259 350.96082 146.47945 351.73917 146.3421 352.54045 146.25052 353.27301 146.20474 353.95983 146.18184 354.55505 146.22763 355.05874 146.3421 355.5395 146.54813 355.92868 146.84574 356.18048 147.21202 356.31786 147.5783 356.31786 147.99038 356.2034 148.37956 355.99735 148.81452 355.72263 149.27237 355.40212 149.75312 355.05874 150.30255 354.66951 150.82907 354.30326 151.30983 353.93695 151.76769 353.59352 152.06529 353.34172 152.29421 353.04409 152.50025 353.34172 152.54604 354.02849 152.52315 354.44059 152.54604 354.82977 152.68339 355.24187 152.86654 355.56238 153.14125 355.81422 153.46175 356.04314 153.75936 356.22627 154.17142 356.36365 154.62928 356.52391 155.13292 356.63837 155.63656 356.72996 156.07152 356.86729 156.43781 357.07334 156.91856 356.77575 157.05591 356.47811 156.98724 356.11181 156.89567 355.65396 156.82699 355.12741 156.68963 354.50926 156.52938 353.82249 156.39202 353.11276 156.32334 352.35727 156.27756 351.62471 156.27756 350.9379 156.32334 350.36554 156.39202 349.97637 156.4836 350.25108 156.09442 350.57159 155.8426</stroke>
</layer>
</page>
</xournal>
```
which crashes. It doesn't crash when replacing the `0` pressure value in `0.58017059 0" capStyle="round">` in the `stroke tool` line by some positive value, e.g. `0.1`

**Comment by bhennion** [comment]
I don't quite see what in #4302 could trigger this. Did you bisect and reached this PR or is this just an estimation?

**Comment by rolandlo** [comment]
Just an estimation. I will bisect now and report back.

**Comment by bhennion** [comment]
I'd say it's probably https://github.com/xournalpp/xournalpp/commit/553588a4cf858d76b357322e40f50f750ca69794

**Comment by rolandlo** [comment]
That makes sense. It's the assertion that fails.
```
xournalpp: ./src/core/view/StrokeViewHelper.cpp:45: xoj::view::StrokeViewHelper::drawWithPressure(cairo_t*, const std::vector<Point>&, const LineStyle&, double)::<lambda(const Point&, const Point&)>: Zusicherung »p.z > 0.0« nicht erfüllt.
```
I will check.

**Comment by bhennion** [comment]
I'm surprised a pressure=0 point could actually be created. I thought we had a minimum for the pressure values. Are the affected files old?

**Comment by rolandlo** [comment]
Hmm, not that old, I annotated it last time on Oct 7 I think. But maybe I used the flatpak (v1.1.1) to edit it.

**Comment by rolandlo** [comment]
Still in v1.1.1 the minimal pressure (0.05 in my case) was already there. The rounding from https://github.com/xournalpp/xournalpp/commit/d370b6f94b284a6b6dfd10a46cfdf5b9220cb421 also does not seem to explain it.

**Comment by rolandlo** [comment]
You were right about the first commit, where it crashes, https://github.com/xournalpp/xournalpp/commit/553588a4cf858d76b357322e40f50f750ca69794. I will check my pressure values in the original .xopp-file more carefully and open other files, too.

**Comment by rolandlo** [comment]
The situation is quite bad. Files I generated with my stylus before January 2021 (i.e. before minimal pressures were introduced in  #2622) can't be opened anymore (they all contain some negative pressure values). Files generated after January 2021 can be opened, with the exception of the file which made me aware of the bug. I believe that I copied some strokes from an old file (annotating basically the same PDF).
Maybe we should take care in the `LoadHandler` of that problem (replace pressures <=0 by the minimal pressure from the settings).

**Comment by bhennion** [comment]
> Maybe we should take care in the `LoadHandler` of that problem (replace pressures <=0 by the minimal pressure from the settings).

Yes, I agree this seems like the best way to go.

**Comment by rolandlo** [comment]
> > Maybe we should take care in the `LoadHandler` of that problem (replace pressures <=0 by the minimal pressure from the settings).
> 
> Yes, I agree this seems like the best way to go.

There are two problems with that idea though:
1) the minimum pressure can also be set to 0.00
2) Neither the LoadHandler nor the Stroke class have access to the settings.

**Comment by bhennion** [comment]
Created a new issue. This is OT here.
